bundles/binding/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/LCNBinding.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lcn;
+
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+
+import org.openhab.binding.lcn.common.*;
+import org.openhab.binding.lcn.connection.*;
+import org.openhab.binding.lcn.input.*;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.types.Command;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+
+/**
+ * Main class of the LCN openHAB binding.
+ * <p>
+ * The binding executes everything on the refresh-thread inside {@link #execute()}.
+ * This assures the required thread-safety.
+ * Input data is received through {@link #connections}.
+ * 
+ * @author Tobias J�ttner
+ *
+ * @param <P> {@link LcnBindingProvider}
+ */
+public class LcnBinding<P extends LcnBindingProvider> extends AbstractActiveBinding<P> implements ManagedService, ConnectionManager.Callback {
```

Comment:
using ManagedService is the "old" variant to read persistent configuration data. Please rather use the new version (see binding.archetype) how the binding should look like then.